hack/verify-github-spelling.sh

```diff
@@ -13,12 +13,15 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-set -o errexit
 set -o nounset
 set -o pipefail
 
-if grep -rn 'Github' /workspace
+find . \( -not -path '*vendor*' -and -not -path '*.git*' -and -not -path '*hack/verify-github-spelling.sh*' \) -type f -exec grep -Hn 'Github' {} +
+
+if [ $? -eq 0 ]
```

Comment:
So we generally don't allow filenames with spaces etc (so no need for `-print0` and no issued with the `for` loop),  so you could do

```sh
failed="false"
for file in $(  find . \( -not -path '*vendor*' -and -not -path '*.git*' -and -not -path '*hack/verify-github-spelling.sh*' \) -type f ); do
    if grep -q 'Github' "${file}"; then
        echo "[ERROR] ${file}: contains mis-spelling 'Github'" 
        failed="true"
    fi
done

if [[ "${failed}" == "true" ]]; then
    exit 1
fi
```

Your discretion -- but I feel like writing Bash that looks like real languages is easier to understand and maintain. @BenTheElder likely has opinions too

Reply:
Looks good to me... updated the script.